rfcs/0017-tx-valid-since/0017-tx-valid-since.md

```diff
@@ -1,335 +1,152 @@
-                        let a = self.epoch_number_with_fraction.to_rational();
-                        let b =
-                            info.block_epoch.to_rational() + epoch_number_with_fraction.to_rational();
-                        if a < b {
-                            return Err(TransactionError::Immature).into());
-                        }
-                    }
-                    Some(SinceMetric::Timestamp(timestamp)) => {
-                        // pass_median_time(current_block) starts with tip block, which is the
-                        // parent of current block.
-                        // pass_median_time(input_cell's block) starts with cell_block_number - 1,
-                        // which is the parent of input_cell's block
-                        let cell_median_timestamp = self.parent_median_time(&info.block_hash);
-                        let current_median_time = self.block_median_time(&self.parent_hash);
-                        if current_median_time < cell_median_timestamp + timestamp {
-                            return Err(TransactionError::Immature).into());
-                        }
-                    }
-                    None => {
-                        return Err(TransactionError::InvalidSince).into());
-                    }
-                }
-            }
-            Ok(())
-        }
-        
-        pub fn verify(&self) -> Result<(), Error> {
-            for (cell_meta, input) in self
-                .rtx
-                .resolved_inputs
-                .iter()
-                .zip(self.rtx.transaction.inputs()) {
-                // ignore empty since
-                let since: u64 = input.since().unpack();
-                if since == 0 {
-                    continue;
-                }
-                let since = Since(since);
-                // check remain flags
-                if !since.flags_is_valid() {
-                    return Err(TransactionError::InvalidSince).into());
-                }
-                
-                // verify time lock
-                self.verify_absolute_lock(since)?;
-                self.verify_relative_lock(since, cell_meta)?;
-            }
-            Ok(())
-        }
-    }
-```
+The new consensus rule allows the transaction input to specify an optional since precondition. CKB nodes must verify the transactions in the commitment zone of a block that all the input since preconditions are fulfilled.
+
+The since precondition locates a unique block in any chain if it is long enough. The precondition is effective before the block, and is fulfilled since the block.
+
+There are three metrics to specify the since precondition:
+
+1. via block number,
+2. epoch number with fraction,
+3. or the median timestamp of the preceding 37 blocks.
+
+All of them strictly increase along the block number.
+
+The precondition is absolute or relative. A relative precondition depends on which block has committed the input.
```

Comment:
"which block has committed the input" sounds ambiguous - it could mean the block commits the input, or the block commits the cell pointed by the input.